src/zodbpickle/_pickle_33.c

```diff
@@ -4,6 +4,28 @@
 PyDoc_STRVAR(pickle_module_doc,
 "Optimized C implementation for the Python pickle module.");
 
+
+#if (PY_VERSION_HEX >= 0x30A00B1) /* 3.10.0b1 */
+#ifndef Py_SIZE
+#define Py_SIZE(ob) (((PyVarObject*)(ob))->ob_size)
+#endif
+#define _PyUnicode_AsStringAndSize PyUnicode_AsUTF8AndSize
+/**
+ * The function ``_PyObject_LookupAttrId`` function replaces the combo of
+ * ``_PyObject_HasAttrId`` followed by ``_PyObject_GetAttrId``; our code isn't
+ * structured to take advantage of that, so for now we throw away the
+ * resulting attribute value and continue to make the extra ``Get`` call.
+ */
+static int _PyObject_HasAttrId(PyObject* obj, void* id)
+{
+    int result;
+    PyObject* attr_val;
```

Comment:
We're not initializing this to `NULL`, but that's not essential. All the error paths through `_PyObject_LookupAttrId` set it to `NULL` on failure, one way or the other, so it's always safe to call `Py_XDECREF`: `attr_val` will never hold garbage at that point. A fair amount of code in CPython itself depends on this behaviour so it's unlikely to change.